samples/file-encryptor/enclave/opensslsrc/encryptor.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <openenclave/enclave.h>
+#include <openssl/aes.h>
+#include <openssl/evp.h>
+#include <string>
+
+#include "../../shared.h"
+
+using namespace std;
+
+class ecall_dispatcher
+{
+  private:
+    EVP_CIPHER_CTX* m_encryption_cipher_ctx;
```

Comment:
This seems the only difference from the mbedtls version. Can we abstract this so that the header can be shared between two implementations?